mesonbuild/build.py

```diff
@@ -362,6 +363,26 @@ def get_project_link_args(self, compiler: 'Compiler', project: str, for_machine:
 
         return link_args.get(compiler.get_language(), [])
 
+@unique
+class IncludeDirsUse(Enum):
+    # 'incdirs', if not absolute paths, are to be used as both src-relative and
+    # build-relative.  Absolute paths are allowed and will just result in one
+    # include dir.
+    src_and_build_relative = 0
+
+    # 'incdirs' will only be used as src-relative (or absolute) include directories.
+    src_relative = 1
+
+    # 'incdirs' are expected to be build-relative include dirs and will error if
+    # absolute paths are used.
+    build_relative = 2
+
+include_dirs_use_kwargs_to_enum: T.Dict[str, IncludeDirsUse] = {
+    'src-and-build-relative': IncludeDirsUse.src_and_build_relative,
+    'src-relative': IncludeDirsUse.src_relative,
+    'build-relative': IncludeDirsUse.build_relative
+}
```

Comment:
This is module constant data, so to fit our style it should be `INCLUDE_DIRS_USE_KWARGS_TO_ENUM`

On the other hand, if we just made the values match the names, then you can do

`IncludeDirsUse[name]` and avoid this, ie `e = IncludeDirsUse['BOTH']`

Reply:
Having names match values would then either be either imposing the internal ALL_CAPS-ness style of the enum values to the parser/interperter's public interface -
```
incdirs = include_directories('headers', use_type: 'BOTH')
```
just to avoid this little translation mapping from the user meson script values to the internal enum, which seems a little unfortunate/clunky... Or we still need an arguably more involved way to translate the user script strings to internal enum vals.  Something like this lambda with conditional generator and 'next'? -
```python
class IncludeDirsUse(Enum):
    BOTH = 'both'
    SRC = 'src'
    BUILD = 'build'

...

     KwargInfo(
        'use_type',
        ...
        validator=in_set_validator(set((e.value for e in IncludeDirsUse))),
        convertor=lambda x: next( (e for e in IncludeDirsUse if x == e.value) )
```

I'm OK with this (and the all caps style) but a str -> enum dict seems marginally preferable to me, but maybe you know of a nicer way to convert from Enum assigned values ('both', 'src', 'build') to an actual enum type (IncludeDirsUse.BOTH, IncludeDirsUse.SRC, etc), where the string values differ from the enum member values, than this 'next' + generator fiddlyness.

Reply:
or you could just do `e = IncludeDirUse[name.upper()]`

Reply:
Thanks. Done.